lib/pact/consumer/server.rb

```diff
@@ -66,7 +66,11 @@ def responsive?
     end
 
     def run_default_server(app, port)
-      require 'rack/handler/webrick'
+      begin
+        require 'rack/handler/webrick'
+      rescue LoadError
+        require 'rackup/handler/webrick'
+      end
       Rack::Handler::WEBrick.run(app, **webrick_opts) do |server|
```

Comment:
if `rackup/handler/webrick` is required in previous rescue, then this needs to be `Rackup::Handler::WEBrick` otherwise it throws a NameError. If it needs to be compatible with `rack` then we need to handle the error doing something like below (though there might be a neater way of doing it)

```
    def run_default_server(app, port)
      begin
        require 'rack/handler/webrick'
      rescue LoadError
        require 'rackup/handler/webrick'
      end
      begin
        Rack::Handler::WEBrick.run(app, **webrick_opts) do |server|
          @port = server[:Port]
        end
      rescue NameError
        Rackup::Handler::WEBrick.run(app, **webrick_opts) do |server|
          @port = server[:Port]
        end
      end
    end
```

Reply:
hey, I assume this is causing you issues as it stands?

happy to consider an additional pull request.

it may be worth tracking your particular problem as a new issue for visibility

Reply:
ive just seen the issue and pr now, thanks!

Reply:
Looking at this now, it looks wrong / incomplete to me too, but we're running it on Rack 3 successfully and it worked when I tested it last year. I haven't looked at the other PR but there must be some case where the existing code works.

Reply:
yep sorry should have followed up with the issue we've created for reference https://github.com/pact-foundation/pact-mock_service/issues/151